src/plugins/es_ui_shared/static/forms/hook_form_lib/components/use_array.ts

```diff
@@ -60,41 +77,26 @@ export const UseArray = ({
   children,
 }: Props) => {
   const isMounted = useRef(false);
-  const uniqueId = useRef(0);
 
   const form = useFormContext();
   const { getFieldDefaultValue } = form;
 
-  const getNewItemAtIndex = useCallback(
-    (index: number): ArrayItem => ({
-      id: uniqueId.current++,
-      path: `${path}[${index}]`,
-      isNew: true,
-    }),
-    [path]
-  );
-
   const fieldDefaultValue = useMemo<ArrayItem[]>(() => {
     const defaultValues = readDefaultValueOnForm
       ? getFieldDefaultValue<unknown[]>(path)
       : undefined;
 
     if (defaultValues) {
-      return defaultValues.map((_, index) => ({
-        id: uniqueId.current++,
-        path: `${path}[${index}]`,
-        isNew: false,
-      }));
+      return defaultValues.map((_, index) => createArrayItem(path, index, false));
     }
 
-    return new Array(initialNumberOfItems).fill('').map((_, i) => getNewItemAtIndex(i));
-  }, [path, initialNumberOfItems, readDefaultValueOnForm, getFieldDefaultValue, getNewItemAtIndex]);
+    return new Array(initialNumberOfItems).fill('').map((_, i) => createArrayItem(path, i));
+  }, [path, initialNumberOfItems, readDefaultValueOnForm, getFieldDefaultValue]);
 
   // Create an internal hook field which behaves like any other form field except that it is not
   // outputed in the form data (when calling form.submit() or form.getFormData())
   // This allow us to run custom validations (passed to the props) on the Array items
-
-  const internalFieldPath = useMemo(() => `${path}__${uuid.v4()}`, [path]);
```

Comment:
We need a predictable field name to be able to update the field values in `updateFieldValues()`